cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -281,13 +246,9 @@ public void getStatus(HttpRequest request, HttpResponder responder,
 
     ProgramType programType = ProgramType.valueOfCategoryName(type);
     Id.Program program = Id.Program.from(namespaceId, appId, programType, id);
-    StatusMap statusMap = getStatus(program);
-    // If status is null, then there was an error
-    if (statusMap.getStatus() == null) {
-      responder.sendString(HttpResponseStatus.valueOf(statusMap.getStatusCode()), statusMap.getError());
-      return;
-    }
-    Map<String, String> status = ImmutableMap.of("status", statusMap.getStatus());
+    ProgramStatus programStatus = getStatus(program);
+
+    Map<String, String> status = ImmutableMap.of("status", programStatus.name());
```

Comment:
Should we create a POJO in ``cdap-proto`` for this object? Just so it becomes easier to add more stuff if needed later? Or do you think its an overkill at this stage?